plugins/destination/duckdb/client/types.go

```diff
@@ -7,10 +7,41 @@ import (
 	"github.com/cloudquery/plugin-sdk/v3/types"
 )
 
-func (c *Client) SchemaTypeToDuckDB(t arrow.DataType) string {
-	switch v := t.(type) {
-	case *arrow.ListType:
-		return c.SchemaTypeToDuckDB(v.Elem()) + "[]"
+type listLike interface {
+	arrow.DataType
+	Elem() arrow.DataType
+}
+
+func plainWrite(dt arrow.DataType) bool {
+	return arrow.TypeEqual(dt, transformTypeForWriting(dt))
+}
+
+func transformTypeForWriting(dt arrow.DataType) arrow.DataType {
```

Comment:
that differs a bit from the `transformType` as duckdb can't write some of our types (namely, json & uuid have to be written as strings))